components/filter/filter.js

```diff
@@ -452,7 +455,7 @@ class Filter extends LocalizeCoreElement(RtlMixin(LitElement)) {
 		const searchValue = e.detail.value.trim();
 		dimension.searchValue = searchValue;
 
-		if (dimension.searchType === 'automatic') {
+		if (dimension.searchType === 'automatic' || searchValue === '') {
```

Comment:
I figured it'd be easier and more performant for the filter to always handle clearing the search, no matter the search-type, since that's just setting everything to `hidden = false`